sdk/eventhub/azure-eventhub/migration_guide.md

```diff
@@ -1,129 +1,179 @@
+We've also merged the functionality from `EventProcessorHost` into `EventHubConsumerClient`. 
 
-In V5, EventHubConsumerClient.receive() calls user callback on_event to process events.
+#### Approachability
+By having a single entry point for sending, the `EventHubProducerClient` helps with the discoverability of the API
+as you can explore all available features for sending events through methods from a single client, as opposed to searching
+through documentation or exploring namespace for the types that you can instantiate.
 
-For example, this code which keeps receiving from a partition in V1:
+Similarly, by having a single entry point for receiving of any type (from single partition, all partitions, or with load balancing and checkpointing features) within Event Hubs, the `EventHubConsumerClient` helps with the discoverability of the API as you can explore all available features for receiving events through methods from a single client, as opposed to searching
+through documentation or exploring namespace for the types that you can instantiate.
 
-```python
-client = EventHubClientAsync.from_connection_string(connection_str, eventhub=EVENTHUB_NAME)
-receiver = client.add_async_receiver(consumer_group="$Default", partition="0", offset=Offset('@latest'))
-try:
-    await client.run_async()
-    logger = logging.getLogger("azure.eventhub")
-    while True:
-        received = await receiver.receive(timeout=5)
-        for event_data in received:
-            logger.info("Message received:{}".format(event_data.body_as_str()))
-finally:
-    await client.stop_async()
-```
+#### Consistency
+We now have methods with similar names, signature and location for sending and receiving.
+This provides consistency and predictability on the various features of the library.
 
-Becomes this in V5:
+### Client constructors
 
+- While we continue to support connection strings when constructing a client, the main difference is when using Azure Active Directory.
+We now use the new [`azure-identity`](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/identity/azure-identity/README.md) library
+to share a single authentication solution between clients of different Azure services.
+
+In v1:
 ```python
-logger = logging.getLogger("azure.eventhub")
-async def on_event(partition_context, event):
-    logger.info("Message received:{}".format(event.body_as_str()))
-    await partition_context.update_checkpoint(event)
+# Authenticate with address (full URI string - optionally includes URL-encoded access policy and key). For example:
+# "amqps://<URL-encoded-SAS-policy>:<URL-encoded-SAS-key>@<namespace>.servicebus.windows.net/<eventhub-name>"
+eventhub_client = EventHubClient(address)
```

Comment:
I was talking with @swathipil offline about this one. If I put myself in a customer shoes and I was using address, I would want to see it acknowledge in the migration guide, but we might do a more explicit sentence to say it's not supported anymore.
But ignoring him, I think will just create even more confusion, given we can't know how popular this constructor was.

Thoughts?

Reply:
I don't think the address was constructed programatically, since the [samples](https://github.com/Azure/azure-sdk-for-python/blob/release/eventhub-v1/sdk/eventhub/azure-eventhubs/examples/test_examples_eventhub.py) don't show this either.
In [this](https://github.com/Azure/azure-sdk-for-python/blob/release/eventhub-v1/sdk/eventhub/azure-eventhubs/examples/batch_send.py) example, it is stored in an env variable, so I seems that it could have been copied from the portal. However, when I check in the portal, I can't find this address anywhere. So, I cannot safely assume that it's copied. 

Based on what Laurent and I discussed, we also don't have telemetry on what the common way used by people was. So, to ensure that customers who used address in v1 are also acknowledged, I could update the address section to this:
```suggestion 
v1:
# Authenticate with address
eventhub_client = EventHubClient(address)
...

v5:
# Address is no longer used for authentication.

#Authenticate with connection string
...
```
By mentioning that address is no longer used in v5, it may not be necessary to include the address format that would be distracting in v1. I can also change this as needed.

Reply:
Lets try that @swathipil